### PR TITLE
Refactor `Profile`, all related tests and calls

### PR DESCRIPTION
**Pull request by ethangraham2001**
Sign-in wasn't working properly because of fields that weren't being serialized properly by firebase. This has been fixed by refactoring profile, and removing unused fields.

- Tests updated and passing
- Firebase interaction for profiles works during manual testing *(for example, likes persist after leaving app and reopening)*
- Fixed some logic in `ItineraryList`
- Sign-in successfully sets profile at launch (if profile doesn't exist prior)
- Updated an image resource that was previously too large, and caused a test to fail due to lack of heap space

## pictures 'r t didn't befall

Screenshot from 2024-05-16 22-10-51

### CODE REVIEW

File: app/src/main/java/com/github/wanderwise_inc/app/viewmodel/ProfileViewModel.kt [hunk 1]
```diff
@@ -47,20 +47,16 @@ class ProfileViewModel(
...
   fun removeLikedItinerary(userUid: String, itineraryUid: String) {
     profileRepository.removeItineraryFromLiked(userUid, itineraryUid)
   }
 
-  suspend fun checkIfItineraryIsLikedByActiveProfile(itineraryUid: String): Boolean {
-    return if (isSignInComplete)
-        profileRepository.checkIfItineraryIsLiked(getUserUid(), itineraryUid)
-    else false
-  }
-
```
Comment (ThomasPFMuller) [inline app/src/main/java/com/github/wanderwise_inc/app/viewmodel/ProfileViewModel.kt:62]: removal of redundant method ✅

**Review by ThomasPFMuller — APPROVED** [review]
LGTM, good clean code refactor 👍

File: app/src/main/java/com/github/wanderwise_inc/app/ui/list_itineraries/ItineraryList.kt [hunk 2]
```diff
@@ -77,15 +79,22 @@ fun ItinerariesListScrollable(
...
         modifier = Modifier.padding(paddingValues).testTag(TestTags.ITINERARY_LIST_SCROLLABLE),
         verticalArrangement = spacedBy(15.dp)) {
           this.items(itineraries, { (iti) -> iti }) { itinerary ->
             val uid = profileViewModel.getUserUid()
-            var isLikedInitially by remember { mutableStateOf(false) }
-            LaunchedEffect(uid) {
+            var isLikedInitially by remember {
+              mutableStateOf(likedItineraries.contains(itinerary.uid))
+            }
+            LaunchedEffect(Unit) {
```
[the remark this answers is not in the document]
Comment (ethangraham2001) [reply app/src/main/java/com/github/wanderwise_inc/app/ui/list_itineraries/ItineraryList.kt:95]: I have not read the docs oops, and I can't remember why I changed it. Let me look into this

[the remark this answers is not in the document]
Comment (ethangraham2001) [reply app/src/main/java/com/github/wanderwise_inc/app/ui/list_itineraries/ItineraryList.kt:95]: Alright so it was launched effect `uid` before and I think I changed it to try and get something working. I'll try again with that key

**Pull request merged**